# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1455733964 18000
#      Wed Feb 17 13:32:44 2016 -0500
# Node ID 1309edb344e114d05e1eaf3a40b44836aafdd22d
# Parent  952854b61ac93de5ea844fec8270d2bf727dd4db
use top_srcdir, top_builddir, abs_top_srcdir or abs_top_builddir only if needed

The build system runs primarily in the top-level directory now, so
unless a Make rule changes the working directory before referencing a
file, it is not necessary to use the abs_top_X or top_X directory
names.

* Makefile.am, build-aux/common.mk, doc/module.mk,
libgui/graphics/module.mk, libgui/module.mk, libgui/src/module.mk,
libinterp/corefcn/module.mk, libinterp/module.mk,
liboctave/cruft/module.mk, liboctave/cruft/ranlib/module.mk,
liboctave/module.mk, liboctave/numeric/module.mk
liboctave/operators/module.mk, scripts/module.mk, src/module.mk:
Only use Use $(top_srcdir), $(top_builddir), $(abs_top_srcdir), and
$(abs_top_builddir) if necessary.

diff --git a/Makefile.am b/Makefile.am
--- a/Makefile.am
+++ b/Makefile.am
@@ -184,17 +184,17 @@ all-local: $(ALL_LOCAL_TARGETS) $(noinst
 	@echo "   make install    - to install (PREFIX=$(prefix))"
 	@echo ""
 
 run-octave: run-octave.in Makefile
 	$(AM_V_GEN)$(do_subst_script_vals) && \
 	chmod a+rx "$@"
 
 octave-config.h: config.h build-aux/mk-octave-config-h.sh
-	$(AM_V_GEN)$(SHELL) $(top_srcdir)/build-aux/mk-octave-config-h.sh $< > $@-t && \
+	$(AM_V_GEN)$(SHELL) $(srcdir)/build-aux/mk-octave-config-h.sh $< > $@-t && \
 	mv $@-t $@
 
 bits/stl_algo.h: build-aux/stl_algo.h-fixed
 	$(AM_V_GEN)$(MKDIR_P) bits && \
 	$(INSTALL_HEADER) $< $@
 
 ## If we aren't trying to fix stl_algo.h, then try to ensure that
 ## there isn't a stray copy sitting in the build tree.
diff --git a/build-aux/common.mk b/build-aux/common.mk
--- a/build-aux/common.mk
+++ b/build-aux/common.mk
@@ -140,19 +140,17 @@ AM_CXXFLAGS = $(XTRA_CXXFLAGS)
 
 BUILD_CXX = @BUILD_CXX@
 BUILD_CXXFLAGS = @BUILD_CXXFLAGS@
 
 # Linker and library flags
 
 LD_CXX = @LD_CXX@
 LD_STATIC_FLAG = @LD_STATIC_FLAG@
-LIBFLAGS = -L$(top_builddir)
 WARN_LDFLAGS = @WARN_LDFLAGS@
-#ALL_LDFLAGS = $(LIBFLAGS) $(LD_STATIC_FLAG) $(CPICFLAG) $(LDFLAGS)
 
 BUILD_LDFLAGS = @BUILD_LDFLAGS@
 
 SH_LD = @SH_LD@
 SH_LDFLAGS = @SH_LDFLAGS@
 
 DL_LD = @DL_LD@
 DL_LDFLAGS = @DL_LDFLAGS@
diff --git a/doc/module.mk b/doc/module.mk
--- a/doc/module.mk
+++ b/doc/module.mk
@@ -287,17 +287,17 @@ doc/interpreter/octave.pdf: $(DOC_IMAGES
 	  rc=$$?; \
 	  $(am__cd) $(srcdir) && \
 	  $$restore $$backupdir/* `echo "./$@" | sed 's|[^/]*$$||'`; \
 	fi; \
 	rm -rf $$backupdir; exit $$rc
 
 doc/interpreter/octave.dvi: doc/interpreter/octave.texi $(srcdir)/doc/interpreter/version-octave.texi | doc/interpreter/$(am__dirstamp)
 	$(AM_V_TEXI2DVI)TEXINPUTS="$(am__TEXINFO_TEX_DIR)$(PATH_SEPARATOR)$$TEXINPUTS" \
-	MAKEINFO='$(MAKEINFO) $(AM_MAKEINFOFLAGS) $(MAKEINFOFLAGS) -I doc/interpreter -I $(abs_top_srcdir)/doc/interpreter' \
+	MAKEINFO='$(MAKEINFO) $(AM_MAKEINFOFLAGS) $(MAKEINFOFLAGS) -I doc/interpreter -I $(srcdir)/doc/interpreter' \
 	$(TEXI2DVI) $(AM_V_texinfo) --build-dir=$(@:.dvi=.t2d) -o $@ $(AM_V_texidevnull) \
 	`test -f 'doc/interpreter/octave.texi' || echo '$(abs_top_srcdir)/'`doc/interpreter/octave.texi
 
 doc/interpreter/octave.pdf: doc/interpreter/octave.texi $(srcdir)/doc/interpreter/version-octave.texi | doc/interpreter/$(am__dirstamp)
 	$(AM_V_TEXI2PDF)TEXINPUTS="$(am__TEXINFO_TEX_DIR)$(PATH_SEPARATOR)$$TEXINPUTS" \
 	MAKEINFO='$(MAKEINFO) $(AM_MAKEINFOFLAGS) $(MAKEINFOFLAGS) -I doc/interpreter -I $(abs_top_srcdir)/doc/interpreter' \
 	$(TEXI2PDF) $(AM_V_texinfo) --build-dir=$(@:.pdf=.t2p) -o $@ $(AM_V_texidevnull) \
 	`test -f 'doc/interpreter/octave.texi' || echo '$(abs_top_srcdir)/'`doc/interpreter/octave.texi
diff --git a/libgui/graphics/module.mk b/libgui/graphics/module.mk
--- a/libgui/graphics/module.mk
+++ b/libgui/graphics/module.mk
@@ -125,26 +125,26 @@ libgui_graphics_libgui_graphics_la_CPPFL
   $(AM_CPPFLAGS) \
   $(FT2_CPPFLAGS) \
   $(FONTCONFIG_CPPFLAGS) \
   $(HDF5_CPPFLAGS) \
   @OCTGUI_DLL_DEFS@ \
   @QT_CPPFLAGS@ \
   -Ilibgui/graphics -I$(srcdir)/libgui/graphics \
   -Isrc -I$(srcdir)/libgui/src \
-  -I$(top_srcdir)/liboctave/cruft/misc \
-  -I$(top_srcdir)/liboctave/array \
-  -I$(top_builddir)/liboctave/numeric -I$(top_srcdir)/liboctave/numeric \
-  -I$(top_builddir)/liboctave/operators -I$(top_srcdir)/liboctave/operators \
-  -I$(top_srcdir)/liboctave/system \
-  -I$(top_srcdir)/liboctave/util \
-  -I$(top_builddir)/libinterp -I$(top_srcdir)/libinterp \
-  -I$(top_builddir)/libinterp/parse-tree -I$(top_srcdir)/libinterp/parse-tree \
-  -I$(top_builddir)/libinterp/corefcn -I$(top_srcdir)/libinterp/corefcn \
-  -I$(top_srcdir)/libinterp/octave-value
+  -I$(srcdir)/liboctave/cruft/misc \
+  -I$(srcdir)/liboctave/array \
+  -Iliboctave/numeric -I$(srcdir)/liboctave/numeric \
+  -Iliboctave/operators -I$(srcdir)/liboctave/operators \
+  -I$(srcdir)/liboctave/system \
+  -I$(srcdir)/liboctave/util \
+  -Ilibinterp -I$(srcdir)/libinterp \
+  -Ilibinterp/parse-tree -I$(srcdir)/libinterp/parse-tree \
+  -Ilibinterp/corefcn -I$(srcdir)/libinterp/corefcn \
+  -I$(srcdir)/libinterp/octave-value
 
 libgui_graphics_libgui_graphics_la_CFLAGS = $(AM_CFLAGS) $(WARN_CFLAGS)
 
 libgui_graphics_libgui_graphics_la_CXXFLAGS = $(AM_CXXFLAGS) $(WARN_CXXFLAGS)
 
 noinst_LTLIBRARIES += libgui/graphics/libgui-graphics.la
 
 libgui_EXTRA_DIST += \
diff --git a/libgui/module.mk b/libgui/module.mk
--- a/libgui/module.mk
+++ b/libgui/module.mk
@@ -43,18 +43,18 @@ libgui_liboctgui_la_SOURCES =
 
 # Dummy C++ source to force C++ linking.
 nodist_EXTRA_libgui_liboctgui_la_SOURCES = libgui/dummy.cc
 
 libgui_liboctgui_la_LIBADD = \
   libgui/qterminal/libqterminal.la \
   libgui/src/libgui-src.la \
   libgui/graphics/libgui-graphics.la \
-  $(top_builddir)/libinterp/liboctinterp.la \
-  $(top_builddir)/liboctave/liboctave.la \
+  libinterp/liboctinterp.la \
+  liboctave/liboctave.la \
   $(LIBOCTGUI_LINK_DEPS)
 
 # Increment these as needed and according to the rules in the libtool manual:
 libgui_liboctgui_current = 1
 libgui_liboctgui_revision = 0
 libgui_liboctgui_age = 0
 
 libgui_liboctgui_version_info = $(libgui_liboctgui_current):$(libgui_liboctgui_revision):$(libgui_liboctgui_age)
diff --git a/libgui/src/module.mk b/libgui/src/module.mk
--- a/libgui/src/module.mk
+++ b/libgui/src/module.mk
@@ -195,27 +195,27 @@ libgui_src_libgui_src_la_CPPFLAGS = \
   $(FONTCONFIG_CPPFLAGS) \
   @OCTGUI_DLL_DEFS@ \
   @QT_CPPFLAGS@ \
   -I$(srcdir)/libgui/qterminal/libqterminal \
   -Ilibgui/src -I$(srcdir)/libgui/src \
   -I$(srcdir)/libgui/src/m-editor \
   -I$(srcdir)/libgui/src/qtinfo \
   -I$(srcdir)/libgui/graphics \
-  -I$(top_srcdir)/liboctave/cruft/misc \
-  -I$(top_srcdir)/liboctave/array \
-  -I$(top_builddir)/liboctave/numeric -I$(top_srcdir)/liboctave/numeric \
-  -I$(top_builddir)/liboctave/operators -I$(top_srcdir)/liboctave/operators \
-  -I$(top_srcdir)/liboctave/system \
-  -I$(top_srcdir)/liboctave/util \
-  -I$(top_builddir)/libinterp -I$(top_srcdir)/libinterp \
-  -I$(top_builddir)/libinterp/parse-tree -I$(top_srcdir)/libinterp/parse-tree \
-  -I$(top_builddir)/libinterp/corefcn -I$(top_srcdir)/libinterp/corefcn \
-  -I$(top_srcdir)/libinterp/octave-value \
-  -I$(top_builddir)/libgnu -I$(top_srcdir)/libgnu
+  -I$(srcdir)/liboctave/cruft/misc \
+  -I$(srcdir)/liboctave/array \
+  -Iliboctave/numeric -I$(srcdir)/liboctave/numeric \
+  -Iliboctave/operators -I$(srcdir)/liboctave/operators \
+  -I$(srcdir)/liboctave/system \
+  -I$(srcdir)/liboctave/util \
+  -Ilibinterp -I$(srcdir)/libinterp \
+  -Ilibinterp/parse-tree -I$(srcdir)/libinterp/parse-tree \
+  -Ilibinterp/corefcn -I$(srcdir)/libinterp/corefcn \
+  -I$(srcdir)/libinterp/octave-value \
+  -Ilibgnu -I$(srcdir)/libgnu
 
 libgui_src_libgui_src_la_CFLAGS = $(AM_CFLAGS) $(WARN_CFLAGS)
 
 libgui_src_libgui_src_la_CXXFLAGS = $(AM_CXXFLAGS) $(WARN_CXXFLAGS)
 
 noinst_LTLIBRARIES += libgui/src/libgui-src.la
 
 libgui_EXTRA_DIST += \
diff --git a/libinterp/corefcn/module.mk b/libinterp/corefcn/module.mk
--- a/libinterp/corefcn/module.mk
+++ b/libinterp/corefcn/module.mk
@@ -2,22 +2,22 @@
 ## These are generated automagically by configure and Perl.
 OPT_HANDLERS = \
   libinterp/corefcn/DASPK-opts.cc \
   libinterp/corefcn/DASRT-opts.cc \
   libinterp/corefcn/DASSL-opts.cc \
   libinterp/corefcn/LSODE-opts.cc \
   libinterp/corefcn/Quad-opts.cc
 
-$(OPT_HANDLERS): libinterp/corefcn/%.cc : $(top_builddir)/liboctave/numeric/%.in
+$(OPT_HANDLERS): libinterp/corefcn/%.cc : liboctave/numeric/%.in
 	$(AM_V_GEN)rm -f $@-t $@ && \
-	$(PERL) $(top_srcdir)/build-aux/mk-opts.pl --opt-handler-fcns $< > $@-t && \
+	$(PERL) $(srcdir)/build-aux/mk-opts.pl --opt-handler-fcns $< > $@-t && \
 	mv $@-t $@
 
-$(OPT_HANDLERS): $(top_srcdir)/build-aux/mk-opts.pl
+$(OPT_HANDLERS): $(srcdir)/build-aux/mk-opts.pl
 
 JIT_INC = \
   libinterp/corefcn/jit-util.h \
   libinterp/corefcn/jit-typeinfo.h \
   libinterp/corefcn/jit-ir.h \
   libinterp/corefcn/pt-jit.h
 
 COREFCN_INC = \
diff --git a/libinterp/module.mk b/libinterp/module.mk
--- a/libinterp/module.mk
+++ b/libinterp/module.mk
@@ -2,29 +2,29 @@ libinterp_EXTRA_DIST =
 
 libinterp_CLEANFILES =
 libinterp_DISTCLEANFILES =
 libinterp_MAINTAINERCLEANFILES =
 
 ## Search local directories before those specified by the user.
 libinterp_liboctinterp_la_CPPFLAGS = \
   @OCTINTERP_DLL_DEFS@ \
-  -I$(top_builddir)/liboctave -I$(top_srcdir)/liboctave \
-  -I$(top_srcdir)/liboctave/cruft/misc \
-  -I$(top_srcdir)/liboctave/array \
-  -I$(top_builddir)/liboctave/numeric -I$(top_srcdir)/liboctave/numeric \
-  -I$(top_builddir)/liboctave/operators -I$(top_srcdir)/liboctave/operators \
-  -I$(top_srcdir)/liboctave/system \
-  -I$(top_srcdir)/liboctave/util \
+  -Iliboctave -I$(srcdir)/liboctave \
+  -I$(srcdir)/liboctave/cruft/misc \
+  -I$(srcdir)/liboctave/array \
+  -Iliboctave/numeric -I$(srcdir)/liboctave/numeric \
+  -Iliboctave/operators -I$(srcdir)/liboctave/operators \
+  -I$(srcdir)/liboctave/system \
+  -I$(srcdir)/liboctave/util \
   -I$(srcdir)/libinterp/octave-value \
   -Ilibinterp -I$(srcdir)/libinterp \
   -I$(srcdir)/libinterp/operators \
   -Ilibinterp/parse-tree -I$(srcdir)/libinterp/parse-tree \
   -Ilibinterp/corefcn -I$(srcdir)/libinterp/corefcn \
-  -I$(top_builddir)/libgnu -I$(top_srcdir)/libgnu \
+  -Ilibgnu -I$(srcdir)/libgnu \
   $(HDF5_CPPFLAGS)
 
 libinterp_liboctinterp_la_CFLAGS = $(AM_CFLAGS) $(WARN_CFLAGS)
 
 libinterp_liboctinterp_la_CXXFLAGS = $(AM_CXXFLAGS) $(WARN_CXXFLAGS)
 
 octlib_LTLIBRARIES += libinterp/liboctinterp.la
 
@@ -131,17 +131,17 @@ DIST_SRC += \
 include libinterp/parse-tree/module.mk
 include libinterp/octave-value/module.mk
 include libinterp/operators/module.mk
 include libinterp/template-inst/module.mk
 include libinterp/corefcn/module.mk
 include libinterp/dldfcn/module.mk
 
 $(srcdir)/libinterp/dldfcn/module.mk: $(srcdir)/libinterp/dldfcn/config-module.sh $(srcdir)/libinterp/dldfcn/config-module.awk $(srcdir)/libinterp/dldfcn/module-files
-	$(AM_V_GEN)$(SHELL) $(srcdir)/libinterp/dldfcn/config-module.sh $(top_srcdir)
+	$(AM_V_GEN)$(SHELL) $(srcdir)/libinterp/dldfcn/config-module.sh $(srcdir)
 
 if AMCOND_ENABLE_DYNAMIC_LINKING
   OCT_FILES = $(DLDFCN_LIBS:.la=.oct)
   DLD_LIBOCTINTERP_LIBADD = libinterp/liboctinterp.la
 else
   OCT_FILES =
   DLD_LIBOCTINTERP_LIBADD =
 endif
@@ -163,17 +163,17 @@ nodist_libinterp_liboctinterp_la_SOURCES
   libinterp/build-env.cc \
   libinterp/build-env-features.cc \
   libinterp/version.h
 
 libinterp_liboctinterp_la_LIBADD = \
   libinterp/octave-value/liboctave-value.la \
   libinterp/parse-tree/libparse-tree.la \
   libinterp/corefcn/libcorefcn.la \
-  $(top_builddir)/liboctave/liboctave.la \
+  liboctave/liboctave.la \
   $(LIBOCTINTERP_LINK_DEPS)
 
 # Increment these as needed and according to the rules in the libtool manual:
 libinterp_liboctinterp_current = 3
 libinterp_liboctinterp_revision = 0
 libinterp_liboctinterp_age = 0
 
 libinterp_liboctinterp_version_info = $(libinterp_liboctinterp_current):$(libinterp_liboctinterp_revision):$(libinterp_liboctinterp_age)
@@ -223,17 +223,17 @@ DLL_CXXDEFS = @OCTINTERP_DLL_DEFS@
 	  $(libinterp_liboctinterp_la_CXXFLAGS) $(CXXFLAGS) \
 	  -DMAKE_BUILTINS $< > $@-t1 && \
 	$(SHELL) $(srcdir)/libinterp/mkdefs $(srcdir)/libinterp $< < $@-t1 > $@-t && \
 	rm -f $@-t1 && \
 	mv $@-t $@
 
 ## Rules to build test files
 
-LIBINTERP_TST_FILES_SRC := $(shell $(SHELL) $(top_srcdir)/build-aux/find-files-with-tests.sh "$(srcdir)" $(ULT_DIST_SRC) $(DLDFCN_SRC))
+LIBINTERP_TST_FILES_SRC := $(shell $(SHELL) $(srcdir)/build-aux/find-files-with-tests.sh "$(srcdir)" $(ULT_DIST_SRC) $(DLDFCN_SRC))
 
 LIBINTERP_TST_FILES := $(addsuffix -tst,$(LIBINTERP_TST_FILES_SRC))
 
 libinterptestsdir := $(octtestsdir)
 
 nobase_libinterptests_DATA = $(LIBINTERP_TST_FILES)
 
 ## Override Automake's rule that forces a .hh extension on us even
@@ -247,17 +247,17 @@ nobase_libinterptests_DATA = $(LIBINTERP
 
 ## build-env.cc must depend on Makefile.
 ## Calling configure may change default/config values.
 ## However, calling configure will also regenerate the Makefiles from
 ## Makefile.am and trigger the rules below.
 libinterp/build-env.cc: libinterp/build-env.in.cc Makefile
 	$(AM_V_GEN)$(do_subst_config_vals)
 
-libinterp/build-env-features.cc: $(top_builddir)/config.h libinterp/build-env-features.sh
+libinterp/build-env-features.cc: config.h libinterp/build-env-features.sh
 	$(AM_V_GEN)rm -f $@-t && \
 	$(SHELL) $(srcdir)/libinterp/build-env-features.sh $< > $@-t && \
 	$(simple_move_if_change_rule)
 
 libinterp/version.h: libinterp/version.in.h Makefile
 	$(AM_V_GEN)rm -f $@-t && \
 	$(SED) < $< \
 	  -e "s|%NO_EDIT_WARNING%|DO NOT EDIT!  Generated automatically from $(<F) by Make.|" \
@@ -298,17 +298,17 @@ libinterp/DOCSTRINGS: | libinterp/.DOCST
 libinterp/.DOCSTRINGS: $(ALL_DEF_FILES) libinterp/gendoc.pl
 	$(AM_V_GEN)rm -f $@-t $@ && \
 	if [ "x$(srcdir)" != "x." ] && [ -f $(srcdir)/libinterp/DOCSTRINGS ] && [ ! -f DOCSTRINGS ]; then \
 		cp $(srcdir)/libinterp/DOCSTRINGS libinterp/DOCSTRINGS; \
 		touch -r $(srcdir)/libinterp/DOCSTRINGS libinterp/DOCSTRINGS; \
 	fi && \
 	$(PERL) $(srcdir)/libinterp/gendoc.pl $(ALL_DEF_FILES) > $@-t && \
 	mv $@-t $@ && \
-	$(SHELL) $(top_srcdir)/build-aux/move-if-change $@ libinterp/DOCSTRINGS && \
+	$(SHELL) $(srcdir)/build-aux/move-if-change $@ libinterp/DOCSTRINGS && \
 	touch $@
 
 OCTAVE_INTERPRETER_TARGETS += \
   libinterp/.DOCSTRINGS
 
 endif
 
 OCTAVE_INTERPRETER_TARGETS += \
diff --git a/liboctave/cruft/module.mk b/liboctave/cruft/module.mk
--- a/liboctave/cruft/module.mk
+++ b/liboctave/cruft/module.mk
@@ -21,17 +21,17 @@ include liboctave/cruft/odepack/module.m
 include liboctave/cruft/ordered-qz/module.mk
 include liboctave/cruft/quadpack/module.mk
 include liboctave/cruft/ranlib/module.mk
 include liboctave/cruft/slatec-err/module.mk
 include liboctave/cruft/slatec-fn/module.mk
 
 define gen-cruft-def
   rm -f $@-t $@ && \
-  $(SHELL) liboctave/cruft/mkf77def $(top_srcdir) $(liboctave_cruft_libcruft_la_SOURCES) > $@-t && \
+  $(SHELL) liboctave/cruft/mkf77def $(srcdir) $(liboctave_cruft_libcruft_la_SOURCES) > $@-t && \
   mv $@-t $@
 endef
 
 ## Special rules for files which must be built before compilation
 liboctave/cruft/cruft.def: $(liboctave_cruft_libcruft_la_SOURCES) liboctave/cruft/mkf77def
 	$(AM_V_GEN)$(gen-cruft-def)
 
 liboctave_DISTCLEANFILES += \
diff --git a/liboctave/cruft/ranlib/module.mk b/liboctave/cruft/ranlib/module.mk
--- a/liboctave/cruft/ranlib/module.mk
+++ b/liboctave/cruft/ranlib/module.mk
@@ -39,17 +39,17 @@ noinst_LTLIBRARIES += liboctave/cruft/ra
 
 liboctave_cruft_ranlib_libranlib_la_SOURCES = $(RANLIB_SRC)
 
 liboctave_cruft_ranlib_libranlib_la_DEPENDENCIES = liboctave/cruft/ranlib/ranlib.def
 
 define gen-ranlib-def
   rm -f $@-t $@ && \
   $(MKDIR_P) liboctave/cruft/ranlib && \
-  $(SHELL) liboctave/cruft/mkf77def $(top_srcdir) $(RANLIB_SRC) > $@-t && \
+  $(SHELL) liboctave/cruft/mkf77def $(srcdir) $(RANLIB_SRC) > $@-t && \
   mv $@-t $@
 endef
 
 ## Special rules for files which must be built before compilation
 ## ranlib directory may not exist in VPATH build; create it if necessary.
 liboctave/cruft/ranlib/ranlib.def: $(RANLIB_SRC) liboctave/cruft/mkf77def
 	$(AM_V_GEN)$(gen-ranlib-def)
 
diff --git a/liboctave/module.mk b/liboctave/module.mk
--- a/liboctave/module.mk
+++ b/liboctave/module.mk
@@ -9,17 +9,17 @@ liboctave_liboctave_la_CPPFLAGS = \
   @OCTAVE_DLL_DEFS@ \
   @CRUFT_DLL_DEFS@ \
   -I$(srcdir)/liboctave/array \
   -I$(srcdir)/liboctave/cruft/misc \
   -Iliboctave/numeric -I$(srcdir)/liboctave/numeric \
   -Iliboctave/operators -I$(srcdir)/liboctave/operators \
   -I$(srcdir)/liboctave/system \
   -I$(srcdir)/liboctave/util \
-  -I$(top_builddir)/libgnu -I$(top_srcdir)/libgnu
+  -Ilibgnu -I$(srcdir)/libgnu
 
 liboctave_liboctave_la_CFLAGS = $(AM_CFLAGS) $(WARN_CFLAGS)
 
 liboctave_liboctave_la_CXXFLAGS = $(AM_CXXFLAGS) $(WARN_CXXFLAGS)
 
 octlib_LTLIBRARIES += liboctave/liboctave.la
 
 BUILT_INCS = \
@@ -62,17 +62,17 @@ include liboctave/util/module.mk
 ## liboctave merely collects a bunch of compiled convenience libraries.
 ## It has no source code itself.
 liboctave_liboctave_la_SOURCES =
 
 # Dummy C++ source to force C++ linking.
 nodist_EXTRA_liboctave_liboctave_la_SOURCES = liboctave/dummy.cc
 
 liboctave_liboctave_la_LIBADD += \
-  $(top_builddir)/libgnu/libgnu.la \
+  libgnu/libgnu.la \
   $(LIBOCTAVE_LINK_DEPS)
 
 # Increment these as needed and according to the rules in the libtool manual:
 liboctave_liboctave_current = 3
 liboctave_liboctave_revision = 0
 liboctave_liboctave_age = 0
 
 liboctave_liboctave_version_info = $(liboctave_liboctave_current):$(liboctave_liboctave_revision):$(liboctave_age)
@@ -89,17 +89,17 @@ liboctave_liboctave_la_LDFLAGS = \
 
 LIBOCTAVE_TST_SRC = \
   $(liboctave_array_libarray_la_SOURCES) \
   $(liboctave_numeric_libnumeric_la_SOURCES) \
   $(liboctave_system_libsystem_la_SOURCES) \
   $(liboctave_util_libutil_la_SOURCES) \
   $(LIBOCTAVE_TEMPLATE_SRC)
 
-LIBOCTAVE_TST_FILES_SRC := $(shell $(SHELL) $(top_srcdir)/build-aux/find-files-with-tests.sh "$(srcdir)" $(LIBOCTAVE_TST_SRC))
+LIBOCTAVE_TST_FILES_SRC := $(shell $(SHELL) $(srcdir)/build-aux/find-files-with-tests.sh "$(srcdir)" $(LIBOCTAVE_TST_SRC))
 
 LIBOCTAVE_TST_FILES := $(addsuffix -tst,$(LIBOCTAVE_TST_FILES_SRC))
 
 liboctavetestsdir := $(octtestsdir)
 
 nobase_liboctavetests_DATA = $(LIBOCTAVE_TST_FILES)
 
 EXTRA_DIST += $(liboctave_EXTRA_DIST)
diff --git a/liboctave/numeric/module.mk b/liboctave/numeric/module.mk
--- a/liboctave/numeric/module.mk
+++ b/liboctave/numeric/module.mk
@@ -94,20 +94,20 @@ NUMERIC_SRC = \
   $(NUMERIC_C_SRC)
 
 LIBOCTAVE_TEMPLATE_SRC += \
   liboctave/numeric/bsxfun-defs.cc
 
 ## Special rules for sources which must be built before rest of compilation.
 $(LIBOCTAVE_OPT_INC) : %.h : %.in
 	$(AM_V_GEN)rm -f $@-t $@ && \
-	$(PERL) $(top_srcdir)/build-aux/mk-opts.pl --opt-class-header $< > $@-t && \
+	$(PERL) $(srcdir)/build-aux/mk-opts.pl --opt-class-header $< > $@-t && \
 	mv $@-t $@
 
-$(LIBOCTAVE_OPT_INC) : $(top_srcdir)/build-aux/mk-opts.pl
+$(LIBOCTAVE_OPT_INC) : $(srcdir)/build-aux/mk-opts.pl
 
 noinst_LTLIBRARIES += liboctave/numeric/libnumeric.la
 
 liboctave_numeric_libnumeric_la_SOURCES = $(NUMERIC_SRC)
 
 liboctave_numeric_libnumeric_la_CPPFLAGS = \
   $(liboctave_liboctave_la_CPPFLAGS) \
   -I$(srcdir)/liboctave/cruft/Faddeeva \
diff --git a/liboctave/operators/module.mk b/liboctave/operators/module.mk
--- a/liboctave/operators/module.mk
+++ b/liboctave/operators/module.mk
@@ -52,17 +52,17 @@ LIBOCTAVE_OPERATORS_INC = \
 ## There are no distributed source files in this directory
 LIBOCTAVE_OPERATORS_SRC =
 
 LIBOCTAVE_TEMPLATE_SRC += \
   liboctave/operators/mx-inlines.cc
 
 ## Special rules for sources which must be built before rest of compilation.
 
-OP_SRCDIR = $(abs_top_srcdir)/liboctave/operators
+OP_SRCDIR = $(srcdir)/liboctave/operators
 
 define run-mk-ops
   rm -f $@-t $@ && \
   $(AWK) -f $(OP_SRCDIR)/mk-ops.awk -v build_file=$(notdir $@) $< > $@-t && \
   mv $@-t $@
 endef
 
 $(BUILT_FULL_MATRIX_OPERATORS_FILES): liboctave/operators/mx-ops liboctave/operators/mk-ops.awk
diff --git a/scripts/module.mk b/scripts/module.mk
--- a/scripts/module.mk
+++ b/scripts/module.mk
@@ -112,17 +112,17 @@ scripts/DOCSTRINGS: | scripts/.DOCSTRING
 scripts/.DOCSTRINGS: $(FCN_FILES) $(GEN_FCN_FILES) scripts/mkdoc.pl Makefile
 	$(AM_V_GEN)rm -f $@-t $@ && \
 	if [ "x$(srcdir)" != "x." ] && [ -f $(srcdir)/scripts/DOCSTRINGS ] && [ ! -f scripts/DOCSTRINGS ]; then \
 		cp $(srcdir)/scripts/DOCSTRINGS scripts/DOCSTRINGS; \
 		touch -r $(srcdir)/scripts/DOCSTRINGS scripts/DOCSTRINGS; \
 	fi && \
 	$(PERL) $(srcdir)/scripts/mkdoc.pl "$(srcdir)" $(FCN_FILES) -- $(GEN_FCN_FILES) > $@-t && \
 	mv $@-t $@ && \
-	$(SHELL) $(top_srcdir)/build-aux/move-if-change $@ scripts/DOCSTRINGS && \
+	$(SHELL) $(srcdir)/build-aux/move-if-change $@ scripts/DOCSTRINGS && \
 	touch $@
 
 OCTAVE_INTERPRETER_TARGETS += \
   scripts/.DOCSTRINGS
 
 endif
 
 check-m-sources:
diff --git a/src/module.mk b/src/module.mk
--- a/src/module.mk
+++ b/src/module.mk
@@ -2,25 +2,25 @@ src_EXTRA_DIST =
 
 src_CLEANFILES =
 src_DISTCLEANFILES =
 src_MAINTAINERCLEANFILES =
 
 ## Search local directories before those specified by the user.
 
 SRC_DIR_CPPFLAGS = \
-  -I$(top_srcdir)/liboctave/array \
-  -I$(top_srcdir)/liboctave/cruft/misc \
-  -I$(top_srcdir)/liboctave/numeric \
-  -I$(top_srcdir)/liboctave/system \
-  -I$(top_srcdir)/liboctave/util \
-  -I$(top_srcdir)/libinterp \
-  -I$(top_builddir)/libinterp/corefcn -I$(top_srcdir)/libinterp/corefcn \
-  -I$(top_srcdir)/src \
-  -I$(top_builddir)/libgnu -I$(top_srcdir)/libgnu
+  -I$(srcdir)/liboctave/array \
+  -I$(srcdir)/liboctave/cruft/misc \
+  -I$(srcdir)/liboctave/numeric \
+  -I$(srcdir)/liboctave/system \
+  -I$(srcdir)/liboctave/util \
+  -I$(srcdir)/libinterp \
+  -Ilibinterp/corefcn -I$(srcdir)/libinterp/corefcn \
+  -I$(srcdir)/src \
+  -Ilibgnu -I$(srcdir)/libgnu
 
 EXTRA_DIST += \
   src/main.in.cc \
   src/mkoctfile.in.cc \
   src/octave-config.in.cc
 
 DISTCLEANFILES += \
   src/main.cc \
@@ -46,24 +46,24 @@ OCTAVE_VERSION_LINKS += src/octave-cli-$
 if AMCOND_BUILD_GUI
   archlib_PROGRAMS += src/octave-gui
   OCTAVE_VERSION_LINKS += src/octave-gui-$(version)$(EXEEXT)
 
   OCTAVE_INTERPRETER_TARGETS += src/octave-gui$(EXEEXT)
 endif
 
 OCTAVE_CORE_LIBS = \
-  $(top_builddir)/libinterp/liboctinterp.la \
-  $(top_builddir)/liboctave/liboctave.la
+  libinterp/liboctinterp.la \
+  liboctave/liboctave.la
 
 nodist_src_octave_SOURCES = src/main.cc
 src_octave_SOURCES = src/display-available.c
 
 src_octave_LDADD = \
-  $(top_builddir)/libgnu/libgnu.la \
+  libgnu/libgnu.la \
   $(X11_LIBS) \
   $(CARBON_LIBS) \
   $(GNULIB_LINK_DEPS)
 
 src_octave_LDFLAGS = \
   $(NO_UNDEFINED_LDFLAG) \
   $(OCTAVE_LINK_OPTS) \
   $(WARN_LDFLAGS)
@@ -96,18 +96,18 @@ src_octave_cli_CPPFLAGS = \
   $(OCTAVE_CPPFLAGS)
 
 src_octave_cli_CXXFLAGS = \
   $(AM_CXXFLAGS) \
   $(WARN_CXXFLAGS)
 
 if AMCOND_BUILD_GUI
   src_octave_gui_SOURCES = src/main-gui.cc
-  OCTAVE_GUI_LIBS = $(top_builddir)/libgui/liboctgui.la
-  OCTAVE_GUI_CPPFLAGS = -I$(top_srcdir)/libgui/src
+  OCTAVE_GUI_LIBS = libgui/liboctgui.la
+  OCTAVE_GUI_CPPFLAGS = -I$(srcdir)/libgui/src
 endif
 
 src_octave_gui_CPPFLAGS = \
   $(SRC_DIR_CPPFLAGS) \
   $(OCTAVE_GUI_CPPFLAGS)
 
 src_octave_gui_LDADD = \
   $(OCTAVE_GUI_LIBS) \
@@ -122,33 +122,33 @@ src_octave_gui_LDFLAGS = \
 src_octave_gui_CXXFLAGS = \
   $(AM_CXXFLAGS) \
   $(WARN_CXXFLAGS)
 
 src_mkoctfile_SOURCES =
 
 nodist_src_mkoctfile_SOURCES = src/mkoctfile.cc
 
-src_mkoctfile_LDADD = $(top_builddir)/libgnu/libgnu.la $(LIBS)
+src_mkoctfile_LDADD = libgnu/libgnu.la $(LIBS)
 
 src_mkoctfile_CPPFLAGS = \
   $(SRC_DIR_CPPFLAGS) \
   $(OCTAVE_CPPFLAGS)
 
 src_mkoctfile_CXXFLAGS = \
   $(AM_CXXFLAGS) \
   $(WARN_CXXFLAGS)
 
 src_octave_config_SOURCES =
 
 nodist_src_octave_config_SOURCES = src/octave-config.cc
 
 src_octave_config_LDADD = \
-  $(top_builddir)/libinterp/corefcn/libcorefcn.la \
-  $(top_builddir)/libgnu/libgnu.la \
+  libinterp/corefcn/libcorefcn.la \
+  libgnu/libgnu.la \
   $(LIBS)
 
 src_octave_config_CPPFLAGS = \
   $(SRC_DIR_CPPFLAGS) \
   $(OCTAVE_CPPFLAGS)
 
 src_octave_config_CXXFLAGS = \
   $(AM_CXXFLAGS) \
@@ -161,27 +161,27 @@ mostlyclean-local: src-mostlyclean-local
 
 if AMCOND_CROSS_TOOLS
 
 ## Building cross mkoctfile.
 
 OCTAVE_CROSS_TOOLS += src/$(host_triplet)-mkoctfile$(BUILD_EXEEXT)
 
 src/$(host_triplet)-mkoctfile$(BUILD_EXEEXT): src/$(host_triplet)-mkoctfile.cc
-	$(BUILD_CXX) -o src/$(host_triplet)-mkoctfile$(BUILD_EXEEXT) -Dgnulib='' -Doctave_idx_type=int $(DEFAULT_INCLUDES) $(BUILD_CXXFLAGS) $(BUILD_LDFLAGS) -I$(top_srcdir)/src src/$(host_triplet)-mkoctfile.cc
+	$(BUILD_CXX) -o src/$(host_triplet)-mkoctfile$(BUILD_EXEEXT) -Dgnulib='' -Doctave_idx_type=int $(DEFAULT_INCLUDES) $(BUILD_CXXFLAGS) $(BUILD_LDFLAGS) -I$(srcdir)/src src/$(host_triplet)-mkoctfile.cc
 
 src/$(host_triplet)-mkoctfile.cc: src/mkoctfile.in.cc Makefile | src/$(octave_dirstamp)
 	$(AM_V_GEN)$(do_subst_cross_config_vals)
 
 ## Building cross octave-config.
 
 OCTAVE_CROSS_TOOLS += src/$(host_triplet)-octave-config$(BUILD_EXEEXT)
 
 src/$(host_triplet)-octave-config$(BUILD_EXEEXT): src/$(host_triplet)-octave-config.cc
-	$(BUILD_CXX) -o src/$(host_triplet)-octave-config$(BUILD_EXEEXT) -Dgnulib='' -Doctave_idx_type=int $(DEFAULT_INCLUDES) $(BUILD_CXXFLAGS) $(BUILD_LDFLAGS) -I$(top_srcdir)/src src/$(host_triplet)-octave-config.cc
+	$(BUILD_CXX) -o src/$(host_triplet)-octave-config$(BUILD_EXEEXT) -Dgnulib='' -Doctave_idx_type=int $(DEFAULT_INCLUDES) $(BUILD_CXXFLAGS) $(BUILD_LDFLAGS) -I$(srcdir)/src src/$(host_triplet)-octave-config.cc
 
 src/$(host_triplet)-octave-config.cc: src/octave-config.in.cc Makefile | src/$(octave_dirstamp)
 	$(AM_V_GEN)$(do_subst_default_vals)
 
 src-mostlyclean-local:
 	-rm -f $(OCTAVE_CROSS_TOOLS)
 
 else
